src/Terrain/terrain.cpp

```diff
@@ -507,13 +507,19 @@ void Terrain::add_all_adjacent(int xyz) {
         if (xyz_ == 13) {
             continue;
         }  //
-        auto [x_, y_, z_] = sop(xyz);
-        auto [xs, ys, zs] = sop(xyz_, 3, 3, 3);
+        auto xyz_center = sop(xyz);
+       	int x_c = xyz_center[0];
+        int y_c = xyz_center[0];
+        int z_c = xyz_center[2];
+        auto xyz_direction = sop(xyz_, 3, 3, 3);
+        int x_d = xyz_direction[0]; // direction to final tile. can be +1, -1, or 0
+        int y_d = xyz_direction[0];
+        int z_d = xyz_direction[2];
```

Comment:
What's wrong with destructuring? Did you set your C++ version?

https://en.cppreference.com/w/cpp/language/structured_binding